### PR TITLE
use single query to getTables

### PR DESCRIPTION
**Pull request by pivnicek**
This should reduce execution time somewhat for the getTables request.

### CODE REVIEW

**Comment by pivnicek** [comment]
ah, must ensure ordering

File: src/Keboola/DbExtractor/Extractor/PgSQL.php [hunk 1]
```diff
@@ -495,8 +458,24 @@ function ($table) {
...
+            case 'r':
+                return "table";
+            case 'v':
+                return "view";
+            case 'm':
+                return "materialized view";
+            case 'f':
+                return "foreign table";
+            default:
+                return null;
```
Comment (odinuv) [inline src/Keboola/DbExtractor/Extractor/PgSQL.php]: let's add a full list https://www.postgresql.org/docs/10/catalog-pg-class.html rather then returning no type 'randomly'

File: tests/Keboola/Extractor/PgsqlTest.php [hunk 2]
```diff
@@ -127,7 +127,7 @@ public function testGetTables(): void
                 array (
                     'name' => 'escaping',
                     'schema' => 'public',
-                    'type' => 'BASE TABLE',
+                    'type' => 'table',
```
Comment (odinuv) [inline tests/Keboola/Extractor/PgsqlTest.php]: isn't this value used used somewhere?

**Comment by odinuv** [comment]
also, have you measured that it reduces the time ?

File: tests/Keboola/Extractor/PgsqlTest.php (hunk 2 above)
Comment (pivnicek) [reply tests/Keboola/Extractor/PgsqlTest.php]: nope

File: src/Keboola/DbExtractor/Extractor/PgSQL.php (hunk 1 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/Extractor/PgSQL.php]: done

**Comment by pivnicek** [comment]
@odinuv the performance test,  https://github.com/keboola/db-extractor-pgsql/blob/master/tests/Keboola/Extractor/PgsqlTest.php#L790, still returns 1 the method, but that's no change.  The changes in this PR do reduce the amount of data transferred though, so the idea is that it will reduce execution time also wrt the network overhead.

File: tests/Keboola/Extractor/PgsqlTest.php (hunk 2 above)
Comment (odinuv) [reply tests/Keboola/Extractor/PgsqlTest.php]: can't we get rid of it completetely? this will reduce the amount of data transferred too

**Comment by odinuv** [comment]
if it returns 1, then we should measure it in ms then i suppose :)

sorry for being obstructive, but if the change is supposed to bring performance gain, then let's see it

**Comment by pivnicek** [comment]
I ran the performance test locally, and with 100,000 tables the execution took 28sec for this new version and the old one took 38 sec.

I don't want to test at this level regularly though because building the test database takes like 1400 sec.

**Pull request merged**